example/route/route.cpp

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CppAlliance/url
+//
+
+#include <boost/url/error.hpp>
+#include <boost/url/segments_encoded.hpp>
+#include <boost/url/segments_encoded_view.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/url_view.hpp>
+#include <boost/url/static_url.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+#include <iostream>
+
+namespace urls = boost::urls;
+namespace fs = boost::filesystem;
+using string_view = urls::string_view;
+
+/// Check if a target matches a prefix
+/**
+    This function checks if the first segments
+    of the target match the corresponding prefix
+    segments.
+
+    @param target Target segments
+    @param prefix Prefix segments
+    @return True if target matches prefix
+ */
+bool match_prefix(
+    urls::segments_view target,
+    urls::segments_view prefix)
+{
+    // Match the prefix segments
+    auto it0 = target.begin();
+    auto end0 = target.end();
+    auto it1 = prefix.begin();
+    auto end1 = prefix.end();
+    while (
+        it0 != end0 &&
+        it1 != end1 &&
+        *it0 == *it1)
+    {
+        ++it0;
+        ++it1;
+    }
+    return it1 == end1;
+}
+
+/// A static route representing files in a directory
+/**
+    A route is a URL logical prefix representing
+    static files in the specified root directory.
+
+    The `match` function returns the corresponding
+    file for a given URL path.
+ */
+class route
+{
+public:
+    /// Constructor
+    route(string_view prefix, fs::path root)
+        : prefix_(urls::parse_uri_reference(prefix).value())
+        , root_(std::move(root))
+    {}
+
+    /// Constructor
+    route(urls::url prefix, fs::path root)
+        : prefix_(std::move(prefix))
+        , root_(std::move(root))
+    {}
+
+    /// Match target URL path with a file
+    /**
+        This function attempts to match the target
+        URL path with the route prefix.
+
+        If the prefix matches, the target is
+        considered to represent a file in the root
+        directory. When that happens, the target
+        prefix is consumed and other segments are
+        appended to the root path.
+
+        The complete file path represented by the
+        target is returned as the output parameter
+        `result`.
+
+        @param target Target URL path
+        @param result Store the corresponding file
+        @return True if target matches the directory
+     */
+    bool match(
+        urls::url_view target,
+        fs::path& result)
+    {
+        if (match_prefix(
+                target.segments(),
+                static_cast<urls::url_view>(prefix_).segments()))
```

Comment:
That's kind of ugly. We probably have to do something about it. 

Requiring `urls::segments` in the function when it only needs and is able to work with `urls::segments_view` doesn't look good.

Some kind of `u.csegments()` to call the constant version of the function, like `cbegin()`/`cend()`? Like `url` and `url_view`, could `segments` maybe inherit from `segments_view`? Or maybe just making `segments` implicitly convertible to `segments_view`?

Reply:
This is ugly but lets just merge this thing and move forward, it doesn't have to be perfect for now. Open an issue!